src/comm/MAVLinkProtocol.cc

```diff
@@ -362,6 +363,10 @@ void MAVLinkProtocol::receiveBytes(LinkInterface* link, QByteArray b)
             // Reset message parsing
             memset(&_status,  0, sizeof(_status));
             memset(&_message, 0, sizeof(_message));
+
+            if (linkPtr.expired()) {
```

Comment:
Can this move up to be right after the emit so that nothing else can sneak in between? Also with a comment as to why it's needed.

Reply:
yup, done.